core/src/test/java/org/apache/calcite/test/enumerable/EnumerableCombineTest.java

```diff
@@ -149,9 +154,62 @@ class EnumerableCombineTest {
               return builder.combine(2).build();
             })
         .returnsUnordered(
-            "QUERY=[{name=Sales}, {name=Marketing}, {name=HR}]",
-            "QUERY=[{empid=100, name=Bill, deptno=10}, {empid=150, name=Sebastian, deptno=10}, "
-                + "{empid=110, name=Theodore, deptno=10}]");
+            "QUERY_0={name=Sales}; QUERY_1={empid=100, name=Bill, deptno=10}",
+            "QUERY_0={name=Marketing}; QUERY_1={empid=150, name=Sebastian, deptno=10}",
+            "QUERY_0={name=HR}; QUERY_1={empid=110, name=Theodore, deptno=10}");
+  }
+
+  /**
+   * Test that two queries selecting from the same table with different sort
+   * orders do not conflict with each other. Each query maintains its own
+   * independent results without one sort order overriding the other.
+   *
+   * <p>Query 1 sorts employees by empid ascending, Query 2 sorts by name descending.
+   * The key verification is that both queries execute independently and produce
+   * their own ordered results.
+   */
+  @Test void testCombineSameTableDifferentSortOrders() {
+    tester(new HrSchema())
+        .withRel(
+            builder -> {
+              // Query 1: SELECT empid, name FROM emps ORDER BY empid ASC
+              builder.scan("s", "emps")
+                  .project(
+                      builder.field("empid"),
+                      builder.field("name"))
+                  .sort(builder.field("empid"));
+
+              // Query 2: SELECT empid, name FROM emps ORDER BY name DESC
+              builder.scan("s", "emps")
+                  .project(
+                      builder.field("empid"),
+                      builder.field("name"))
+                  .sort(builder.desc(builder.field("name")));
+
+              // Combine both queries
+              return builder.combine(2).build();
+            })
+        .returns(resultSet -> {
+          try {
+            int rowCount = 0;
+            while (resultSet.next()) {
+              rowCount++;
+              // Verify both columns have values (no cross-contamination)
+              Object query0 = resultSet.getObject("QUERY_0");
+              Object query1 = resultSet.getObject("QUERY_1");
+              if (query0 == null || query1 == null) {
```

Comment:
Would it be better to use AssertThat for positive validation? I don't understand why the throw new AssertionError approach is used here.